### PR TITLE
Do not use deprecated reinit function.

### LINKED ISSUE #3292: do not use deprecated reinit() in trilinos_vector.h:721
```
include/deal.II/lac/trilinos_vector.h:721:7: warning: 'reinit' is deprecated [-Wdeprecated-declarations]
      reinit (vector_partitioner(), v);
      ^
```

happens for example for

```
TrilinosWrappers::MPI::Vector t_src;
Vector<double> tmp;
...
t_src = tmp;
```

### PR DESCRIPTION
**Pull request by Rombur**
fix #3292

### CODE REVIEW

File: include/deal.II/lac/trilinos_vector.h [hunk 1]
```diff
@@ -718,7 +718,14 @@ namespace TrilinosWrappers
...
         }
 
-      reinit (vector_partitioner(), v);
+      const Epetra_Map &map = vector_partitioner();
+      const int size = map.NumMyElements();
+
+      // Need to copy out values, since the deal.II might not use doubles, so
+      // that a direct access is not possible.
+      for (int i=0; i<size; ++i)
+        (*vector)[0][i] = v(gid(map,i));
```
Comment (bangerth) [inline include/deal.II/lac/trilinos_vector.h]: Is `gid` a global function in Trilinos? Isn't there a corresponding member function of the `Epetra_Map` we could call here to make the code more obvious?

Comment (Rombur) [inline include/deal.II/lac/trilinos_vector.h]: `gid` is one of our own [function](https://github.com/Rombur/dealii/blob/ef52c69409b6ee3e7d0233e7937172db9c5d6a7f/include/deal.II/lac/trilinos_vector.h#L51). The function in trilinos is either `map.GID` or `map.GID64` if using 64 bits.

Comment (bangerth) [inline include/deal.II/lac/trilinos_vector.h]: Ah, I see :-)

**Comment by bangerth** [comment]
I think the test failures are still a left-over from the changes to quoting MPI tests (should be fixed on master by now). So I will merge.

**Pull request merged**